src/main/java/io/strimzi/kafka/bridge/SourceBridgeEndpoint.java

```diff
@@ -80,11 +82,25 @@ protected void handleClose() {
      */
     protected void send(KafkaProducerRecord<String, byte[]> krecord, Handler<AsyncResult<RecordMetadata>> handler) {
 
-        log.debug("Sending to Kafka on topic {} at partition {} and key {}", krecord.topic(), krecord.partition(), krecord.key());
         if (handler == null) {
             this.producerSettledMode.write(krecord);
         } else {
-            this.producerUnsettledMode.write(krecord, handler);
+            log.debug("Sending to topic " + krecord.topic() + " at partition {}", krecord.partition());
+            this.producerUnsettledMode.partitionsFor(krecord.topic(), part -> {
+                if (part.succeeded()) {
+                    if (krecord.partition() == null) {
+                        //TODO should not to be!
+                        this.producerUnsettledMode.write(krecord, handler);
```

Comment:
So you say this is applying round-robin somehow and I can just delete this comment? @ppatierno

Reply:
When the partition is null and the key is null, the producer uses round robin but it's not a logic we have to add it's in the Kafka producer. The code here hasn't to be changed. You should try to catch the right exception for these kinds of errors. Please try and see the right exception raised by the producer.

Reply:
In that case, everything is ok :)